### ISSUE TITLE
[BUG] Timber\Post->terms() throws error if 'fields' parameter has value other than 'all'

### ISSUE THREAD

**Issue by herrschuessler** [timber/timber#2802]
### Expected Behavior

I'm running the example code found  at https://timber.github.io/docs/v2/getting-started/a-post-archive/#an-example-related-posts
```PHP
$context = Timber::context();

$post = $context['post'];

$context['related_posts'] = Timber::get_posts([
    'post_type' => 'post',
    'posts_per_page' => 3,
    'no_found_rows' => true,
    'orderby' => 'date',
    'order' => 'DESC',
    'post__not_in' => [$post->ID],
    'category__in' => $post->terms([
        'taxonomy' => 'category',
        'fields' => 'ids',
    ]),
]);
```
It should get a post collection posts from the same category.

### Actual behavior

```
Fatal error: Uncaught Error: Timber\Factory\TermFactory::build(): Argument #1 ($term) must be of type WP_Term, int given
in /timber/src/Factory/TermFactory.php on line 136
```

The issues seems to come from the 'fields' parameter in the category query. Setting the param to any other value than 'all' will result in the same error.

### Steps to reproduce behavior

Use Timber 2 rc.1 and run the above code

### What version of Timber are you using?

Timber 2 rc.1

### What version of WordPress are you using?

WordPress 6.3.1

### What version of PHP are you using?

PHP 8.1

### How did you install Timber?

Installed or updated Timber through Composer

**Comment by jrathert** [comment]
Hm, as far as I understand the code, the problem is actually caused by this function in `TermFactory.php`:

```
protected function from_wp_term_query(WP_Term_Query $query): iterable
{
    return \array_map([$this, 'build'], $query->get_terms());
}

```
`$query->get_terms()` (actually: `WP_Term_Query::get_terms()` might - depending on the `fields` argument inside it - return very different things (`WP_Term[]|int[]|string[]|string`), but the code expects it returns a `ẀP_term[]` in all cases (which it only does, if `fields` is not set or set to `all` or  `all_with_object_id`). So I guess this requires quite some work here (and I do not see how the whole code/approach could work if someone sets `fields` to, e.g., `count`)?

See: https://developer.wordpress.org/reference/classes/wp_term_query/get_terms/

**Comment by gchtr** [comment]
Resolved in #2806.
